src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/QueryOrRequestParameterValueResolver.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentResolver;
+
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Attribute\QueryParameter;
+use Symfony\Component\HttpKernel\Attribute\RequestParameter;
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+
+final class QueryOrRequestParameterValueResolver implements ValueResolverInterface
+{
+    /**
+     * @return array<mixed>
+     */
+    public function resolve(Request $request, ArgumentMetadata $argument): array
+    {
+        $attribute = $argument->getAttributesOfType(QueryParameter::class)[0] ?? $argument->getAttributesOfType(RequestParameter::class)[0] ?? null;
+        if (null === $attribute) {
+            return [];
+        }
+
+        $inputBag = $attribute instanceof QueryParameter ? $request->query : $request->request;
+
+        $name = $attribute->name ?? $argument->getName();
+        if (false === $inputBag->has($name)) {
+            if ($argument->isNullable() || $argument->hasDefaultValue()) {
+                return [];
+            }
+
+            throw new BadRequestException(sprintf('Missing "%s" parameter "%s".', $attribute instanceof QueryParameter ? 'query' : 'request', $name));
+        }
+
+        if ($argument->isVariadic()) {
+            return $inputBag->all($name);
+        }
+
+        if ('array' === $argument->getType()) {
+            return [$inputBag->all($name)];
+        }
+
+        $value = $inputBag->get($name);
+
+        if ('string' === $argument->getType()) {
+            return [$value];
+        }
+
+        if ('int' === $argument->getType()) {
+            if (!is_numeric($value)) {
+                throw new BadRequestException(sprintf('"%s" parameter "%s" must be an integer.', $attribute instanceof QueryParameter ? 'Query' : 'Request', $name));
+            }
+
+            return [(int) $value];
+        }
+
+        if ('float' === $argument->getType()) {
+            if (!is_numeric($value)) {
+                throw new BadRequestException(sprintf('"%s" parameter "%s" must be a float.', $attribute instanceof QueryParameter ? 'Query' : 'Request', $name));
+            }
+
+            return [(float) $value];
+        }
+
+        if ('bool' === $argument->getType()) {
+            return [filter_var($value, \FILTER_VALIDATE_BOOLEAN)];
+        }
+
+        return [];
```

Comment:
shouldn't we throw in this case? the attribute says we should do something

Reply:
What would you suggest to throw?

Reply:
Now throwing a `LogicException`, similar as what's being done in:
https://github.com/symfony/symfony/blob/f3722d58568e996bf79fe322ec9100a01e71c9d9/src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/BackedEnumValueResolver.php#L81

Reply:
We should not throw a LogicException for something coming from the query string, because that query string is controlled by the user, not by the dev. On the other hand, an unexpected type in the request _attributes_ indicates a bug in the code of the project, that the dev can fix.

Reply:
Which is the case here: the exception would be thrown only if the dev made a mistake, isn't it?

Reply:
Fixed.